third_party/amd/lib/Dialect/TritonAMDGPU/IR/Dialect.cpp

```diff
@@ -241,4 +241,23 @@ UpcastMXFPOp::deduceOutputType(TypedValue<RankedTensorType> inputTensor,
   return RankedTensorType::get(newShape, outputElemType, newVEncoding);
 }
 
+LogicalResult BufferLoadToLocalOp::verify() {
+  auto other = getOther();
+  if (!other)
+    return success();
+
+  // Other should have the same data type as the ptr and the same shape/layout
```

Comment:
Why manually verify it instead of encoding the rules with ODS like `amdgpu.buffer_load` op?

Reply:
Because other's type depends on $ptr and $offset, other loads can get the type directly from result. I changed it to be in ODS but `TypesMatchWith` it's not a great fit. Maybe you can point me to a better interface (if there is any)?